### PR TITLE
Add a `days_since` implementation to `Date` struct.

### LINKED ISSUE #2: Calculations between dates.
I'm writing a program that does math with dates. 
It would be useful to have some kind of method that can take two dates and return a duration in units.
Something like [this method](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/temporal/TemporalUnit.html#between(java.time.temporal.Temporal,java.time.temporal.Temporal)).

Would you be willing to accept pull requests for such a feature?

### PR DESCRIPTION
**Pull request by wildwestrom**
Very simple, it's a start.

Fixes #2.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@
 //! ```
 //! use gregorian::{Date, Month::*, Year, YearMonth};
 //!
-//! assert!(Year::new(2020).has_leap_day(), true);
+//! assert!(Year::new(2020).has_leap_day(), "{}", true);
```
Comment (de-vri-es) [inline src/lib.rs]:
```suggestion
//! assert!(Year::new(2020).has_leap_day());
```

The `true` actually shouldn't be there :)

**Comment by de-vri-es** [comment]
Thanks for making the PR!

I'm still contemplating if this should return a new struct called `Days`. I'm inclined to go that way, so we can implement `Date: Add<Days>` in the future.

On the other hand, it would mean a lot of work to add math operators for the type to make it convenient to work with.

**Comment by wildwestrom** [comment]
How would that look? Should it be in a separate file? 
What should this `Days` struct be used for?
- Durations?
- Days of the Month?
- Both?
- Neither?

Should it be aware of the gregorian calendar somehow, or does it not really need to?
Would it be preferable to refactor a lot of the current code to use the `Days` struct?

**Comment by de-vri-es** [comment]
> How would that look? Should it be in a separate file?
> What should this `Days` struct be used for?

If we add a `Days` struct, it should only be used for durations, not for day-of-month. You can't sanely add days-of-month to a date.

> Should it be aware of the gregorian calendar somehow, or does it not really need to?

I don't believe that a duration in number of days needs to be aware of the calendar.

> Would it be preferable to refactor a lot of the current code to use the `Days` struct?

Hmm, we do already have `days_since_year_zero()` which just returns an i32. It makes sense to return `Days` there too. However, that would be backwards incompatible, so I prefer not to do that now.

Maybe we should just go for the simplest solution after all, and stick with `i32`.

**Comment by wildwestrom** [comment]
We're in the very early stages, so API changes are to be expected, no?

**Comment by de-vri-es** [comment]
True. But I'm also still in doubt myself if we should or shouldn't have a `Days` struct. I'd certainly be willing to give it a try, but I don't want to ask you to implement something and then decide I don't want it.

Unless you're OK with seeing it as an experiment to get a feeling if `Days` is nicer to work with than `i32`.

**Comment by wildwestrom** [comment]
We can push this in for now, but I'll experiment on my own and see if I like using `Days`.

**Comment by de-vri-es** [comment]
Ok, sounds good. I will hold off on a release until we've come to a decision on the return type.

**Comment by de-vri-es** [comment]
Thanks again for the PR! I'll also experiment a bit with `Days` to see if I like it.

**Comment by de-vri-es** [comment]
Well, it's released now. Plain integer it is :p

**Pull request merged**